cores/esp8266/flash_hal.cpp

```diff
@@ -28,141 +28,27 @@ extern "C" {
 #include "c_types.h"
 #include "spi_flash.h"
 }
-/*
- spi_flash_read function requires flash address to be aligned on word boundary.
- We take care of this by reading first and last words separately and memcpy
- relevant bytes into result buffer.
-
-alignment:       012301230123012301230123
-bytes requested: -------***********------
-read directly:   --------xxxxxxxx--------
-read pre:        ----aaaa----------------
-read post:       ----------------bbbb----
-alignedBegin:            ^
-alignedEnd:                      ^
-*/
```

Comment:
These comments shouldn't be dropped, but should instead be moved to where the logic is now implemented

Reply:
This logic is dropped, since it was not true, address does not need to be aligned (apart from page boundary case that we talked)